docs/docs/api/index.md

```diff
@@ -0,0 +1,82 @@
+# Mathesar's API
+
+Mathesar has an API available at `/api/rpc/v0/` which follows the [JSON-RPC](https://www.jsonrpc.org/specification) spec version 2.0. It has been built primarily for consumption by the Mathesar front end but theoretically could be used to build automation workflows and Mathesar integrations.
+
+## Caveats
+
+- The Mathesar API is **not yet stable**. If you build logic that depends on it, be mindful that future Mathesar versions will likely bring breaking changes to the API without warning or notice.
+
+- This API _documentation_ is nascent and somewhat rough. We encourage you to report any docs inaccuracies via GitHub [issues](https://github.com/mathesar-foundation/mathesar/issues).
+
+- A small subset of functionality in Mathesar still relies on a legacy REST API which is gradually being phased out. It is not documented here.
+
+## Usage
+
+### Requests
+
+To use an RPC function:
+
+- Call it with a dot path starting from its root path.
+- Always use named parameters.
+- Ensure that your request includes HTTP headers for valid session IDs, as well as CSRF cookies and tokens.
+
+!!! example
+
+    To call function `tables.list` from the Tables section of this page, you'd send something like:
```

Comment:
There's a discrepancy in calling the RPC methods "functions" on this page, but then referring to them as "Methods" in the sidebar. It would be nice to unify that language.

Additionally, it would be nice if this page referenced the methods page and gave some context, for example: "See the methods page for avaliable RPC methods". 

To put it another way: How do we help users translate this formatting from the python methods page:

`add(*, database_id, user_id, configured_role_id, **kwargs)`

into the equivalent rpc call formatting:

```json
{
  "jsonrpc": "2.0",
  "id": 234,
  "method": "tables.list",
  "params": {
    "schema_oid": 47324,
    "database_id": 1
  }
}
```

perhaps one hand-picked example on the index page would be enough?

Reply:
Sounds good. Can you implement this? I would like to stick with "methods" because that's what the RPC spec uses.